### ISSUE TITLE
background_color field of Frame is unused

### ISSUE THREAD

**Issue by jdm** [servo/webrender#3674]
This is causing regressions in Servo's tests when we upgrade to the latest revision.

**Comment by jdm** [comment]
It appears to have been removed in https://github.com/servo/webrender/commit/a5efadbadf87f59f25b8d70b44ff3713211f1990.

**Comment by gw3583** [comment]
This is certainly a regression, and it looks like we could probably re-add the semantic to use frame.bg_color if renderer.bg_color is None.

However, there's some subtle issues that the bg_color code introduce which complicate some of the optimizations we're currently working on (related to document splitting and picture caching).

Gecko doesn't use the bg color at all, and instead adds a full-size color rectangle at the root of the display list. This gives greater control over what things look like during checkerboarding, as well as simplifying things internally for WR.

Could we consider switching Servo to do the same and removing frame.bg_color from the interface? This would also reduce the change of future regressions, since it would be the same method that Gecko uses.

**Comment by jdm** [comment]
Yes, it would be good to increase the commonality between Servo and Gecko's approaches.
